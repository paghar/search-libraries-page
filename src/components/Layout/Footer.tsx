import Facebook from "../misc/svg/Facebook";
import Twitter from "../misc/svg/Twitter";
import Instagram from "../misc/svg/Instagram";
import Linkedin from "../misc/svg/Linkedin";
import GitHub from "../misc/svg/GitHub";
import Email from "../misc/svg/Email";
import Phone from "../misc/svg/Phone";

const Footer = () => {
  return(
    <footer className="w-full py-6 px-4 bg-pink-700"> 
      <div className="mx-auto max-w-7xl flex flex-col items-center text-white">
        <div className="w-full flex flex-col md:flex-row flex-wrap justify-between items-center">
          {/* :SOCIALMEDIA */}
          <div className="py-4 px-2 flex">
            <Facebook className="w-6 h-6 mr-1"/>
            <Twitter className="w-6 h-6 mr-1"/>
            <Instagram className="w-6 h-6 mr-1"/>
            <Linkedin className="w-6 h-6 mr-1"/>
            <GitHub className="w-6 h-6 mr-1"/>          
          </div>
          {/* :NAVIGATION */}
          <nav className="py-4 px-2 flex flex-wrap justify-center list-none">
            <li className="p-3 hover:font-semibold">Team</li>
            <li className="p-3 hover:font-semibold">Terms and Conditions</li>
            <li className="p-3 hover:font-semibold">Privacy Policy</li>
            <li className="p-3 hover:font-semibold">API</li>            
          </nav>
          {/* :CONTACT */}
          <div className="px-2 flex flex-col">
            {/* Email */}
            <p className="flex text-xs font-medium tracking-wide">
              <Email className="mr-1 h-5 w-5"/>            
              <a href="#email">fancytailwind@paradise.com</a>
            </p>
            {/* Phone */}
            <p className="flex text-xs  font-bold">
              <Phone className="mr-1 h-5 w-5" />              
              <span>209-217-2459</span>
            </p>
          </div>
        </div>

        {/* :COPYRIGHT */}
        <p className="pt-10 text-sm text-center">&copy;2024, Fancy ,React, next.js All Rights Reserved.</p>

      </div>
    </footer>

  );
};

export default Footer;