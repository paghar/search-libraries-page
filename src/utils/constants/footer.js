export const footerItems = [   
  {value: 1, text: "Team"},
  {value: 2, text: "Terms and Conditions"},
  {value: 3, text: "Privacy Policy"},  
  {value: 4, text: "API"},  
];