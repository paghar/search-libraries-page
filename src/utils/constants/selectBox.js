export const sortItems = [   
  {value: 1, text: "Stars"},
  {value: 2, text: "Ranks"},
  {value: 3, text: "Dependents"},  
  {value: 4, text: "Latest Release"},
  {value: 5, text: "Contributions"},
  {value: 6, text: "Newest"},  
];