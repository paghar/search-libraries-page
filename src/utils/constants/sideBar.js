export const sideBarItems = [   
  {value: 1, text: "Home"},
  {value: 2, text: "Creating Packages"},
  {value: 3, text: "API"},  
  {value: 4, text: "Configuration"},  
  {value: 5, text: "Pluggable Resolvers"},  
  {value: 6, text: "Tools"},  
  {value: 7, text: "About"},  
];