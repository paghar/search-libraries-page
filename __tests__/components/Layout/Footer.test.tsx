import React from "react";
import {render} from "@testing-library/react";
import Footer from "../../../src/components/Layout/Footer";
import {footerItems} from "../../../src/utils/constants/footer";

describe("Footer Component", () => {
  const {container} = render(<Footer/>);
  it("Renders Snapshot Footer", () => {    
    expect(container).toMatchSnapshot();
  });
});


describe("Footer Component", () => {
  test("renders footer with correct social media icons", () => {
    const {getByTestId} = render(<Footer />);
    
    const facebookIcon = getByTestId("facebook-icon");
    const twitterIcon = getByTestId("twitter-icon");
    const instagramIcon = getByTestId("instagram-icon");
    const linkedinIcon = getByTestId("linkedin-icon");
    const githubIcon = getByTestId("github-icon");

    expect(facebookIcon).toBeInTheDocument();
    expect(twitterIcon).toBeInTheDocument();
    expect(instagramIcon).toBeInTheDocument();
    expect(linkedinIcon).toBeInTheDocument();
    expect(githubIcon).toBeInTheDocument();
  });

  test("renders navigation items correctly", () => {
    const {getByText} = render(<Footer />);
    footerItems.forEach(item => {
      const textElement = getByText(item.text);
      expect(textElement).toBeInTheDocument();
    });
  });
  
});