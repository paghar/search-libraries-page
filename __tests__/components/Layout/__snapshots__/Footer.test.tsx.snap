// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Renders Snapshot Footer 1`] = `
<div>
  <footer
    class="w-full py-6 px-4 bg-pink-700"
  >
    <div
      class="mx-auto max-w-7xl flex flex-col items-center text-white"
    >
      <div
        class="w-full flex flex-col md:flex-row flex-wrap justify-between items-center"
      >
        <div
          class="py-4 px-2 flex"
        >
          <svg
            class="w-6 h-6 mr-1"
            data-testid="facebook-icon"
            fill="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 2h-3a5 5 0 00-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 011-1h3z"
            />
          </svg>
          <svg
            class="w-6 h-6 mr-1"
            data-testid="twitter-icon"
            fill="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M23 3a10.9 10.9 0 01-3.14 1.53 4.48 4.48 0 00-7.86 3v1A10.66 10.66 0 013 4s-4 9 5 13a11.64 11.64 0 01-7 2c9 5 20 0 20-11.5a4.5 4.5 0 00-.08-.83A7.72 7.72 0 0023 3z"
            />
          </svg>
          <svg
            class="w-6 h-6 mr-1"
            data-testid="instagram-icon"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <rect
              height="20"
              rx="5"
              ry="5"
              width="20"
              x="2"
              y="2"
            />
            <path
              d="M16 11.37A4 4 0 1112.63 8 4 4 0 0116 11.37zm1.5-4.87h.01"
            />
          </svg>
          <svg
            class="w-6 h-6 mr-1"
            data-testid="linkedin-icon"
            fill="currentColor"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="0"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 8a6 6 0 016 6v7h-4v-7a2 2 0 00-2-2 2 2 0 00-2 2v7h-4v-7a6 6 0 016-6zM2 9h4v12H2z"
              stroke="none"
            />
            <circle
              cx="4"
              cy="4"
              r="2"
              stroke="none"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="w-6 h-6 mr-1"
            data-testid="github-icon"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              clip-rule="evenodd"
              d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <nav
          class="py-4 px-2 flex flex-wrap justify-center list-none"
        >
          <li
            class="p-3 hover:font-semibold"
          >
            Team
          </li>
          <li
            class="p-3 hover:font-semibold"
          >
            Terms and Conditions
          </li>
          <li
            class="p-3 hover:font-semibold"
          >
            Privacy Policy
          </li>
          <li
            class="p-3 hover:font-semibold"
          >
            API
          </li>
        </nav>
        <div
          class="px-2 flex flex-col"
        >
          <p
            class="flex text-xs font-medium tracking-wide"
          >
            <svg
              class="mr-1 h-5 w-5"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.003 5.884L10 9.882l7.997-3.998A2 2 0 0016 4H4a2 2 0 00-1.997 1.884z"
              />
              <path
                d="M18 8.118l-8 4-8-4V14a2 2 0 002 2h12a2 2 0 002-2V8.118z"
              />
            </svg>
            <a
              href="#email"
            >
              fatemeh.paghar@outlook.com
            </a>
          </p>
          <p
            class="flex text-xs  font-bold"
          >
            <svg
              class="mr-1 h-5 w-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 5a2 2 0 012-2h3.28a1 1 0 01.948.684l1.498 4.493a1 1 0 01-.502 1.21l-2.257 1.13a11.042 11.042 0 005.516 5.516l1.13-2.257a1 1 0 011.21-.502l4.493 1.498a1 1 0 01.684.949V19a2 2 0 01-2 2h-1C9.716 21 3 14.284 3 6V5z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <span>
              176-44544529
            </span>
          </p>
        </div>
      </div>
      <p
        class="pt-10 text-sm text-center"
      >
        Â©2024, Fancy Tailwind,React,Next.js,TypeScript All Rights Reserved.
      </p>
    </div>
  </footer>
</div>
`;
